lib/src/command/outdated.dart

```diff
@@ -509,34 +508,40 @@ Future<void> _outputHuman(
           row.upgradable != row.resolvable)
       .length;
 
-  if (upgradable != 0) {
-    if (upgradable == 1) {
-      log.message('\n1 upgradable dependency is locked (in pubspec.lock) to '
-          'an older version.\n'
-          'To update it, use `pub upgrade`.');
-    } else {
-      log.message(
-          '\n$upgradable upgradable dependencies are locked (in pubspec.lock) '
-          'to older versions.\n'
-          'To update these dependencies, use `pub upgrade`.');
+  if (lockFileExists) {
+    if (upgradable != 0) {
+      if (upgradable == 1) {
+        log.message('\n1 upgradable dependency is locked (in pubspec.lock) to '
+            'an older version.\n'
+            'To update it, use `pub upgrade`.');
+      } else {
+        log.message(
+            '\n$upgradable upgradable dependencies are locked (in pubspec.lock) '
+            'to older versions.\n'
+            'To update these dependencies, use `pub upgrade`.');
+      }
     }
-  }
 
-  if (notAtResolvable != 0) {
-    if (notAtResolvable == 1) {
-      log.message('\n1 dependency is constrained to a '
-          'version that is older than a resolvable version.\n'
-          'To update it, edit pubspec.yaml.');
-    } else {
-      log.message('\n$notAtResolvable  dependencies are constrained to '
-          'versions that are older than a resolvable version.\n'
-          'To update these dependencies, edit pubspec.yaml.');
+    if (notAtResolvable != 0) {
+      if (notAtResolvable == 1) {
+        log.message('\n1 dependency is constrained to a '
+            'version that is older than a resolvable version.\n'
+            'To update it, edit pubspec.yaml.');
+      } else {
+        log.message('\n$notAtResolvable  dependencies are constrained to '
+            'versions that are older than a resolvable version.\n'
+            'To update these dependencies, edit pubspec.yaml.');
+      }
     }
-  }
 
-  if (notAtResolvable == 0 && upgradable == 0 && rows.isNotEmpty) {
-    log.message('\nDependencies are all on the latest resolvable versions.'
-        '\nNewer versions, while available, are not mutually compatible.');
+    if (notAtResolvable == 0 && upgradable == 0 && rows.isNotEmpty) {
+      log.message('\nDependencies are all on the latest resolvable versions.'
+          '\nNewer versions, while available, are not mutually compatible.');
+    }
```

Comment:
You've made these conditional on `lockFileExists`, right?

But do we really need to... I see that it makes sense to not advocate people run `pub upgrade` (because nothing is held back by `pubspec.lock`).. But newer resolvable dependencies that are held back by `pubspec.yaml` is something we could mention.

Reply:
Done